theano/gpuarray/dnn.py

```diff
@@ -3746,19 +3746,33 @@ def local_dnn_reduction(node):
             node.op.acc_dtype == 'float64'):
         return
 
-    if node.op.pre_scalar_op is not None:
-        # Might want to handle absmax, avg, norm1, norm2 here
+    def _identity(a):
+        return a
+
+    def _square(a):
+        return GpuElemwise(theano.scalar.basic.sqr)(a)
+
+    scal = node.op.scalar_op.name
+    post = _identity
+
+    if (isinstance(node.op.pre_scalar_op, theano.scalar.basic.Sqr) and
+            isinstance(node.op.scalar_op, theano.scalar.basic.Add)):
+        scal = 'norm2'
+        post = _square
+    elif node.op.pre_scalar_op is not None:
```

Comment:
While at it, can you add the case for norm1? This way, we shouldn't need to make such fix in the futur. If you can't do that today, I think we should merge this PR, but if it can be done quickly, it can be added to this PR.